### PR TITLE
swb: fix over FieldWithBarrier

### PR DESCRIPTION
**Pull request by jianchun**
Made some batch replacement errors before in "FunctionBody.h". Only direct
FunctionBody fields need "FieldWithBarrier" (FunctionBody always uses
software write barrier). Change others to "Field".

### CODE REVIEW

File: lib/Runtime/Base/FunctionBody.h [hunk 1]
```diff
@@ -1779,12 +1779,12 @@ namespace Js
...
     private:
-        FieldWithBarrier(const byte*) m_functionBytes;
-        FieldWithBarrier(ByteCodeCache*) m_cache;
-        FieldWithBarrier(const char16 *) m_displayName;  // Optional name
-        FieldWithBarrier(uint) m_displayNameLength;
-        FieldWithBarrier(uint) m_displayShortNameOffset;
-        FieldWithBarrier(NativeModule *) m_nativeModule;
+        Field(const byte*) m_functionBytes;
+        Field(ByteCodeCache*) m_cache;
+        Field(const char16 *) m_displayName;  // Optional name
```
Comment (leirocks) [inline lib/Runtime/Base/FunctionBody.h]: this sometimes is recycler allocated, sometimes not, should use FieldWithBarrier.
check m_displayNameIsRecyclerAllocated and FunctionProxy::SetDisplayName() for detail

Comment (leirocks) [inline lib/Runtime/Base/FunctionBody.h]: or maybe not. looks SetDisplayName() is taking care of the barrier bit. did you run stress test?

Comment (jianchun) [inline lib/Runtime/Base/FunctionBody.h]: This is not "FunctionBody". It is "DeferDeserializeFunctionInfo" which is not allocated in write barrier memory.

Comment (leirocks) [inline lib/Runtime/Base/FunctionBody.h]: ah, OK

**Pull request merged**